agent/lm_agent/reconciliation.py

```diff
@@ -144,55 +145,146 @@ async def filter_cluster_update_licenses(licenses_to_update: List) -> List:
-                try:
-                    minimum_value = config.features[feature].get("total")
-                    LicenseManagerFeatureConfigurationIncorrect.require_condition(
-                        minimum_value,
-                        f"The configuration for {feature} is incorrect. Please include the total amount of licenses.",
-                    )
-                except AttributeError:
-                    # Fallback to get the total from the old feature format
-                    minimum_value = config.features[feature]
-                server_type = config.license_server_type
-                break
-        slurm_used = await get_tokens_for_license(product_feature + "@" + server_type, "Used")
-        if slurm_used is None:
-            slurm_used = 0
-        new_quantity = license_total - license_used - bookings_sum + slurm_used
-        if new_quantity > license_total:
-            new_quantity = license_total
-        if new_quantity < minimum_value:
-            new_quantity = minimum_value
         product, feature = product_feature.split(".")
-        update_resource = await sacctmgr_modify_resource(product, feature, new_quantity)
-        if update_resource:
-            logger.info("Slurmdbd updated successfully.")
-        else:
-            logger.info("Slurmdbd update unsuccessful.")
-    return r
+        server_used = license_data["license_used"]
+        cluster_name = await get_cluster_name()
+        cluster_booking_sum = await get_booking_sum_for_cluster(cluster_name)
+        other_cluster_booking_sum = await get_booking_sum_for_other_clusters(cluster_name)
+
+        # Get license configuration from backend
+        config_id = await get_config_id_from_backend(product_feature)
+        config = await backend_client.get(f"/lm/api/v1/config/{config_id}")
+        config = config.json()
+
+        license_server_type = config["license_server_type"]
+        # Use feature name to get total and limit from feature data in the license config
+        try:
+            # Get total from new feature format
+            total = config["features"][feature].get("total")
+            LicenseManagerFeatureConfigurationIncorrect.require_condition(
+                total,
+                f"The configuration for {feature} is incorrect. Please include the total amount of licenses.",
+            )
+        except AttributeError:
+            # Fallback to get the total from the old feature format
+            total = config["features"][feature]
+
+        try:
+            # Get limit from new feature format. If not specified, use the total as the limit
+            limit = config["features"][feature].get("limit", total)
+        except AttributeError:
+            # Fallback to use the total as the limit for the old feature format
+            limit = total
+
+        # Get license usage from the cluster
+        slurm_used = await get_tokens_for_license(product_feature + "@" + license_server_type, "Used")
```

Comment:
Probably best to use an f-string here if only for consistency (see line 281).

Reply:
Yes, this was a left over from the previous version of the code.